src/main/java/com/xceptance/xlt/report/mergerules/AbstractPatternRequestFilter.java

```diff
@@ -32,26 +32,14 @@
 public abstract class AbstractPatternRequestFilter extends AbstractRequestFilter
 {
     /**
-     * Cache the expensive stuff but with little sync overhead
+     * Cache the expensive stuff, we are a per thread instance. Can be empty!
      */
-    private ThreadLocal<LRUFastHashMap<CharSequence, Matcher>> cache = new ThreadLocal<LRUFastHashMap<CharSequence, Matcher>>()
-    {
-        @Override
-        protected LRUFastHashMap<CharSequence, Matcher> initialValue()
-        {
-            return new LRUFastHashMap<>(cacheSize);
-        }
-    };
-
-    /**
-     * The size of the cache, because different filters have different demands
-     */
-    private final int cacheSize;
+    private final LRUFastHashMap<CharSequence, Matcher> cache;
 
     /**
      * Just a place holder for a NULL
      */
-    private static final Matcher NULL = Pattern.compile(".*").matcher("null");
+    public static final Matcher NULL = Pattern.compile(".*").matcher("null");
```

Comment:
Why public now?

Reply:
It was supposed to be used outside that that did not work. Changed.